runtime-attach-core/src/main/java/io/opentelemetry/contrib/attach/AgentFileProvider.java

```diff
@@ -5,24 +5,25 @@
 
 package io.opentelemetry.contrib.attach;
 
-import io.opentelemetry.javaagent.OpenTelemetryAgent;
 import java.io.File;
 import java.io.IOException;
 import java.io.InputStream;
-import java.net.URL;
 import java.nio.file.Files;
 import java.nio.file.Path;
-import java.security.CodeSource;
 
 final class AgentFileProvider {
 
-  static File getAgentFile() {
+  private final String agentJarResourceName;
 
-    verifyExistenceOfAgentJarFile();
+  AgentFileProvider(String agentJarResourceName) {
+    this.agentJarResourceName = agentJarResourceName;
+  }
+
+  File getAgentFile() {
 
     Path tempDirPath = createTempDir();
 
-    Path tempAgentJarPath = createTempAgentJarFile(tempDirPath);
+    Path tempAgentJarPath = createTempAgentJarFileIn(tempDirPath);
```

Comment:
what is the reason for creating the temp folder?

Reply:
> what is the reason for creating the temp folder?

Does [this](https://github.com/open-telemetry/opentelemetry-java-contrib/pull/380#discussion_r915681240) answer the question?